src/test/java/org/folio/tags/support/APITest.java

```diff
@@ -0,0 +1,116 @@
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import org.folio.spring.FolioModuleMetadata;
+import org.folio.spring.integration.XOkapiHeaders;
+import org.folio.tenant.domain.dto.TenantAttributes;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+@ContextConfiguration(initializers = APITest.DockerPostgreDataSourceInitializer.class)
+@AutoConfigureMockMvc
+@Testcontainers
+public abstract class APITest {
+
+  protected static final String TOKEN =
+    "eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJkaWt1X2FkbWluIiwidXNlcl9pZCI6IjFkM2I1OGNiLTA3YjUtNWZjZC04YTJhLTNjZTA2YTBlYjkwZiIsImlhdCI6MTYxNjQyMDM5MywidGVuYW50IjoiZGlrdSJ9.2nvEYQBbJP1PewEgxixBWLHSX_eELiBEBpjufWiJZRs";
+  protected static final String TENANT = "test";
+  protected static final String USER_ID = "77777777-7777-7777-7777-777777777777";
+
+  private static final ObjectMapper OBJECT_MAPPER;
+
+  static PostgreSQLContainer<?> postgreDBContainer = new PostgreSQLContainer<>("postgres:12-alpine");
+
+  static {
+    postgreDBContainer.start();
+    OBJECT_MAPPER = new ObjectMapper().setSerializationInclusion(JsonInclude.Include.NON_NULL)
+      .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+      .configure(DeserializationFeature.ACCEPT_SINGLE_VALUE_AS_ARRAY, true);
+  }
+
+  @Autowired
+  protected MockMvc mockMvc;
+  @Autowired
+  protected DatabaseHelper databaseHelper;
+
+  @BeforeAll
+  static void beforeAll(@Autowired MockMvc mockMvc) {
+    setUpTenant(mockMvc);
+  }
+
+  @AfterAll
+  static void afterAll() {
+    postgreDBContainer.stop();
+  }
+
+  @SneakyThrows
+  protected static void setUpTenant(MockMvc mockMvc) {
+    mockMvc.perform(post("/_/tenant").content(asJsonString(new TenantAttributes().moduleTo("mod-tags")))
+      .headers(defaultHeaders())
+      .contentType(APPLICATION_JSON))
+      .andExpect(status().isOk());
+  }
+
+  @SneakyThrows
+  public static String asJsonString(Object value) {
+    return OBJECT_MAPPER.writeValueAsString(value);
+  }
+
+  public static HttpHeaders defaultHeaders() {
+    final HttpHeaders httpHeaders = new HttpHeaders();
+
+    httpHeaders.setContentType(APPLICATION_JSON);
+    httpHeaders.add(XOkapiHeaders.TENANT, TENANT);
+    httpHeaders.add(XOkapiHeaders.TOKEN, TOKEN);
+    httpHeaders.add(XOkapiHeaders.USER_ID, USER_ID);
+
+    return httpHeaders;
+  }
+
+  @TestConfiguration
+  static class TestConfig {
+
+    @Bean
+    public DatabaseHelper databaseHelper(JdbcTemplate jdbcTemplate, FolioModuleMetadata moduleMetadata) {
+      return new DatabaseHelper(moduleMetadata, jdbcTemplate);
+    }
+  }
+
+  static class DockerPostgreDataSourceInitializer implements
+    ApplicationContextInitializer<ConfigurableApplicationContext> {
+
+    @Override
+    public void initialize(ConfigurableApplicationContext applicationContext) {
```

Comment:
Can we replace this configuration with @DynamicPropertySource? For example:

```
 @DynamicPropertySource
  static void postgresqlProperties(DynamicPropertyRegistry registry) {
    registry.add("spring.datasource.url", postgreSQLContainer::getJdbcUrl);
    registry.add("spring.datasource.password", postgreSQLContainer::getPassword);
    registry.add("spring.datasource.username", postgreSQLContainer::getUsername);
  }
```

Reply:
We use spring version >= 2.2.6 and we can simplify setting for DB.
https://wiki.folio.org/pages/viewpage.action?spaceKey=FOLIJET&title=How+to+Use+Testcontainers+in+Integration+Tests#HowtoUseTestcontainersinIntegrationTests-BasicapplicationintegrationtestwithJUnit5andSpringBoot%3E=2.2.6